examples/dashboard/src/lib.rs

```diff
@@ -86,44 +140,8 @@ impl Component for Model {
             Msg::FetchData(format, binary) => {
                 self.fetching = true;
                 let task = match format {
-                    Format::Json => {
-                        let callback = self.link.send_back(
-                            move |response: Response<Json<Result<DataFromFile, Error>>>| {
-                                let (meta, Json(data)) = response.into_parts();
-                                println!("META: {:?}, {:?}", meta, data);
-                                if meta.status.is_success() {
-                                    Msg::FetchReady(data)
-                                } else {
-                                    Msg::Ignore // FIXME: Handle this error accordingly.
-                                }
-                            },
-                        );
-                        let request = Request::get("/data.json").body(Nothing).unwrap();
-                        if binary {
-                            self.fetch_service.fetch_binary(request, callback)
-                        } else {
-                            self.fetch_service.fetch(request, callback)
-                        }
-                    }
-                    Format::Toml => {
-                        let callback = self.link.send_back(
-                            move |response: Response<Toml<Result<DataFromFile, Error>>>| {
-                                let (meta, Toml(data)) = response.into_parts();
-                                println!("META: {:?}, {:?}", meta, data);
-                                if meta.status.is_success() {
-                                    Msg::FetchReady(data)
-                                } else {
-                                    Msg::Ignore // FIXME: Handle this error accordingly.
-                                }
-                            },
-                        );
-                        let request = Request::get("/data.toml").body(Nothing).unwrap();
-                        if binary {
-                            self.fetch_service.fetch_binary(request, callback)
-                        } else {
-                            self.fetch_service.fetch(request, callback)
-                        }
-                    }
+                    Format::Json => Model::format_json(self, binary),
+                    Format::Toml => Model::format_toml(self, binary),
```

Comment:
Can you please change to `self.fetch_json(binary)` and `self.fetch_toml(binary)`?

Reply:
Done 😊